packages/core/src/api/sync/index.ts

```diff
@@ -516,6 +530,98 @@ class Sync {
       }
     });
   }
+
+  private async handleInboxItems(inboxItems: SyncInboxItem[]) {
+    for (const item of inboxItems) {
+      try {
+        if (await this.db.notes.exists(item.id)) {
+          this.logger.info("Inbox item already exists, skipping.", {
+            inboxItemId: item.id
+          });
+          continue;
+        }
+
+        const inboxKeys = await this.db.user.getInboxKeys();
+        if (!inboxKeys) continue;
```

Comment:
Do this once outside the loop. I know we have caching but still no point in calling it on each iteration.